agent/acs/handler/attach_eni_handler.go

```diff
@@ -107,6 +109,13 @@ func (attachENIHandler *attachENIHandler) handleSingleMessage(message *ecsacs.At
 	if err != nil {
 		seelog.Warnf("Failed to ack request with messageId: %s, error: %v", aws.StringValue(message.MessageId), err)
 	}
+
+	// Check if this is a duplicate message
+	if _, ok := attachENIHandler.state.ENIByMac(aws.StringValue(message.ElasticNetworkInterfaces[0].MacAddress)); ok {
```

Comment:
minor: can you assign `message.ElasticNetworkInterfaces[0].MacAddress` to a local var? Makes it simpler to read the code.